runtime/src/main/java/org/corfudb/runtime/Restore.java

```diff
@@ -136,23 +146,31 @@ private void enableCompaction() {
     private void restore() throws IOException {
         if (restoreMode == RestoreMode.FULL) {
             clearAllTables();
+        } else if (restoreMode == RestoreMode.PARTIAL_NAMESPACE) {
+            clearTablesInNamespace(namespace);
         }
 
         long startTime = System.currentTimeMillis();
         int restoredTableCount = 0;
-        for (String tableBackup : tableBackups) {
-            log.info("start restoring table {}", tableBackup);
-            if (restoreMode == RestoreMode.PARTIAL_TAGGED && !isTableTagged(tableBackup)) {
-                log.info("skip restoring table {} since it doesn't have requires_backup_support tag", tableBackup);
+        for (TableBackupFileName tableBackupFileName : tableBackupFileNames) {
+            log.info("start restoring table {}", tableBackupFileName);
+            if (restoreMode == RestoreMode.PARTIAL_TAGGED && !isTableTagged(tableBackupFileName)) {
+                log.info("skip restoring table {} since it doesn't have requires_backup_support tag", tableBackupFileName);
+                continue;
+            }
+
+            if (restoreMode == RestoreMode.PARTIAL_NAMESPACE &&
```

Comment:
Just for clarification: In restore it looks like we can do either PARTIAL_TAGGED only or PARTIAL_NAMESPACE only. But in backup we can do an intersection of taggedTablesOnly and namespace. Not sure of the use cases but is this intended?

Reply:
Currently there is no detailed requirements for the namespace related APIs. What you mentioned could be requested in the future and will be implemented upon requests.